src/test/java/com/codeus/winter/config/DefaultBeanFactoryTest.java

```diff
@@ -115,6 +123,20 @@ void testInitializeManyBeansWithDependencyInReverseOrder() {
         assertEquals(beanA, beanB.getBeanA());
         assertEquals(beanA, beanC.getBeanA());
         assertEquals(beanB, beanC.getBeanB());
+        BeanD beanD = factory.getBean(BeanD.class);
```

Comment:
This test case covers exactly the case when we have pending dependencies. I prefer to leave this test as it is and add separate tests for collections for each type of collection